src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java

```diff
@@ -28,151 +29,156 @@
+			Secret sourceSecret = getAuthenticatedOpenShiftClient().secrets()
+					.inNamespace(buildConfig.getMetadata().getNamespace())
+					.withName(buildConfig.getSpec().getSource().getSourceSecret().getName()).get();
+			return upsertCredential(sourceSecret, namespace, secretName);
+
+		}
+		return null;
+	}
+
+	/**
+	 * Inserts or creates a Jenkins Credential for the given Secret
+	 */
+	public static synchronized String upsertCredential(Secret secret) throws IOException {
+		if (secret != null) {
+			ObjectMeta metadata = secret.getMetadata();
+			if (metadata != null) {
+				return upsertCredential(secret, metadata.getNamespace(), metadata.getName());
+			}
+		}
+		return null;
+	}
+
+	private static String upsertCredential(Secret secret, String namespace, String secretName) throws IOException {
+		String id = null;
+		if (secret != null) {
+			Credentials creds = secretToCredentials(secret);
+			id = secretName(namespace, secretName);
+			Credentials existingCreds = lookupCredentials(id);
+			final SecurityContext previousContext = ACL.impersonate(ACL.SYSTEM);
+			try {
+				CredentialsStore s = CredentialsProvider.lookupStores(Jenkins.getActiveInstance()).iterator().next();
+				if (existingCreds != null) {
+					s.updateCredentials(Domain.global(), existingCreds, creds);
+				} else {
+					s.addCredentials(Domain.global(), creds);
+				}
+			} finally {
+				SecurityContextHolder.setContext(previousContext);
+			}
+		}
+		return id;
+	}
+
+	// getCurrentToken returns the ServiceAccount token currently selected by
+	// the user. A return value of empty string
+	// implies no token is configured.
+	public static String getCurrentToken() {
+		String credentialsId = GlobalPluginConfiguration.get().getCredentialsId();
+		if (credentialsId.equals("")) {
+			return "";
+		}
+
+		OpenShiftToken token = CredentialsMatchers
+				.firstOrNull(
+						CredentialsProvider.lookupCredentials(OpenShiftToken.class, Jenkins.getActiveInstance(),
+								ACL.SYSTEM, Collections.<DomainRequirement>emptyList()),
+						CredentialsMatchers.withId(credentialsId));
+
+		if (token != null) {
+			return token.getToken();
+		}
+
+		return "";
+	}
+
+	private static Credentials lookupCredentials(String id) {
+		return CredentialsMatchers
+				.firstOrNull(
+						CredentialsProvider.lookupCredentials(Credentials.class, Jenkins.getActiveInstance(),
+								ACL.SYSTEM, Collections.<DomainRequirement>emptyList()),
+						CredentialsMatchers.withId(id));
+	}
+
+	private static String secretName(String namespace, String name) {
+		String watchNamespace = null;
+		GlobalPluginConfiguration config = GlobalPluginConfiguration.get();
+		if (config != null) {
+			watchNamespace = config.getNamespace();
+		}
+		// if we only watch a single namespace and this secret is in that namespace then
+		// lets avoid the namespace prefix in the name
+		if (watchNamespace != null && namespace.equals(watchNamespace)) {
+			return name;
```

Comment:
@rupalibehera - I believe @bparees 's comment here got lost in the shuffle of other changes.  Special casing this seem problematic if the user changes configuration and adds multiple namespaces later on.

Reply:
So I will remove this for now then let's return namespace-name always then